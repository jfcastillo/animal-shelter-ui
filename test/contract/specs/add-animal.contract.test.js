import { provider } from './init-pact';
import { AnimalController } from "../../../controllers";
import { Matchers } from "@pact-foundation/pact";

const animal = {
    name: "Lisa",
    breed: "Criolla",
    gender: "Female",
    vaccinated: true
}
describe('Given an Animal Service', () => {
    // beforeAll(async () => {
    //     await provider.setup();
    // });
    describe('When a request to add a new animal is made', () => {
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                state: 'create animal',
                uponReceiving: 'a request to add a new animal',
                withRequest: {
                    method: 'POST',
                    path: '/animals',
                    body: animal,
                },
                willRespondWith: {
                    status: 201,
                    body: Matchers.like(animal)
                }
            });
        });
        it("Then it should return the right data", async () => {
            const response = await AnimalController.register(animal);
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });
    });
    afterAll(async () => {
        await provider.finalize();
    })
});