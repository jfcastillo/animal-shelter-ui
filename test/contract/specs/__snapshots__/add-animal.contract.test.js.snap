// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an Animal Service When a request to add a new animal is made Then it should return the right data 1`] = `
Object {
  "breed": "Criolla",
  "gender": "Female",
  "name": "Lisa",
  "vaccinated": true,
}
`;
